debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/ValueTypeHelper.java

```diff
@@ -177,11 +178,17 @@ private J9ClassPointer findJ9ClassInFlattenedClassCacheWithFieldSigImpl(Structur
 		public J9ClassPointer[] findNestedClassHierarchy(J9ClassPointer containerClazz, String[] nestingHierarchy) throws CorruptDataException {
 			J9ClassPointer[] resultClasses = new J9ClassPointer[nestingHierarchy.length + 1];
 			J9ClassPointer clazz = containerClazz;
-			resultClasses[0] = containerClazz;
+			int index = 0;
 
-			for (int i = 0; i < nestingHierarchy.length; i++) {
-				clazz = findJ9ClassInFlattenedClassCacheWithFieldName(clazz, nestingHierarchy[i]);
-				resultClasses[i + 1] = clazz;
+			if (Pattern.matches("\\[([1-9]\\d*|0)\\]", nestingHierarchy[0])) {
```

Comment:
Same comment about pattern: don't be unnecessarily restrictive.

Reply:
What guarantees that `nestingHierarchy.length > 0`?

Reply:
This function is only called within `PrintObjectFieldsHelper.java` if flatobject is true and the condition for flatobject is `((null != nestingHierarchy) && (nestingHierarchy.length > 0))`